# Distribuição de Renda de 2020 no Brasil

Este notebook analisa alguns aspectos da distribuição de renda no Brasil.

Os dados da  Receita Federal estão disponíveis através do portal de [Dados Abertos do Governo Federal](https://dados.gov.br/dados/conjuntos-dados/distribuio-de-renda).

### Perguntas respondidas

### Perguntas em aberto

- Qual a tributação que cada centil paga?
- Como é a distribuição em outros países? É local ou sistemático essa diferença?

## Código

In [1]:
import numpy as np
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go

In [2]:
df = pd.read_csv("data/distribuicao-renda.csv", sep=";")
df.head()

Ano-calendário Ente Federativo Centil  Quantidade de Contribuintes  \
0            2006          BRASIL      1                      241.563   
1            2006          BRASIL      2                      241.563   
2            2006          BRASIL      3                      241.562   
3            2006          BRASIL      4                      241.563   
4            2006          BRASIL      5                      241.562   

  Rendimentos Tributaveis - Limite Superior da RTB do Centil [R$ milhões]  \
0                                                NaN                        
1                                                NaN                        
2                                                NaN                        
3                                                NaN                        
4                                                NaN                        

  Rendimentos Tributaveis - Soma da RTB do Centil [R$ milhões]  \
0                                                NaN             
1                                                NaN             
2                                                NaN             
3                                                NaN             
4                                                NaN             

  Rendimentos Tributaveis - RTB Acumulada do Centil [R$ milhões]  \
0                                                NaN               
1                                                NaN               
2                                                NaN               
3                                                NaN               
4                                                NaN               

  Rendimentos Tributaveis - Média da RTB do Centil [R$]  \
0                                                NaN      
1                                                NaN      
2                                                NaN      
3                                                NaN      
4                                                NaN      

  Rendimentos Sujeitos à Tribut. Exclusiva [R$ milhões]  \
0                                             235,61      
1                                             208,74      
2                                             219,96      
3                                             257,01      
4                                             249,88      

  Rendimentos Isentos - Lucros e dividendos [R$ milhões]  ...  \
0                                             481,27      ...   
1                                             483,44      ...   
2                                             459,87      ...   
3                                             481,93      ...   
4                                             464,23      ...   

  Despesas Dedutíveis - Instrução [R$ milhões]  \
0                                          NaN   
1                                          NaN   
2                                          NaN   
3                                          NaN   
4                                          NaN   

  Despesas Dedutíveis - Médicas [R$ milhões]  \
0                                        NaN   
1                                        NaN   
2                                        NaN   
3                                        NaN   
4                                        NaN   

  Despesas Dedutíveis - Pensão Alimentícia [R$ milhões]  \
0                                                NaN      
1                                                NaN      
2                                                NaN      
3                                                NaN      
4                                                NaN      

  Despesas Dedutíveis - Livro-Caixa [R$ milhões] Imposto Devido [R$ milhões]  \
0                                            NaN                        0,16   
1                                            NaN                        0,22   
2        

In [3]:
df20 = df[df["Ano-calendário"] == 2020]
df20 = df20.drop(["Ano-calendário"], axis=1)
df20.head()

Ente Federativo Centil  Quantidade de Contribuintes  \
43260          BRASIL      1                      316.349   
43261          BRASIL      2                      316.348   
43262          BRASIL      3                      316.349   
43263          BRASIL      4                      316.348   
43264          BRASIL      5                      316.349   

      Rendimentos Tributaveis - Limite Superior da RTB do Centil [R$ milhões]  \
43260                                                NaN                        
43261                                                NaN                        
43262                                                NaN                        
43263                                                NaN                        
43264                                                NaN                        

      Rendimentos Tributaveis - Soma da RTB do Centil [R$ milhões]  \
43260                                                NaN             
43261                                                NaN             
43262                                                NaN             
43263                                                NaN             
43264                                                NaN             

      Rendimentos Tributaveis - RTB Acumulada do Centil [R$ milhões]  \
43260                                                NaN               
43261                                                NaN               
43262                                                NaN               
43263                                                NaN               
43264                                                NaN               

      Rendimentos Tributaveis - Média da RTB do Centil [R$]  \
43260                                                NaN      
43261                                                NaN      
43262                                                NaN      
43263                                                NaN      
43264                                                NaN      

      Rendimentos Sujeitos à Tribut. Exclusiva [R$ milhões]  \
43260                                           2.292,22      
43261                                           2.774,79      
43262                                           2.735,77      
43263                                           2.314,48      
43264                                           2.378,27      

      Rendimentos Isentos - Lucros e dividendos [R$ milhões]  \
43260                                           4.971,46       
43261                                           5.276,95       
43262                                           4.733,55       
43263                                           4.838,44       
43264                                           5.007,19       

      Rendimentos Isentos - Rendim. Sócio/Titular ME/EPP Opt SIMPLES [R$ milhões]  \
43260                                           2.199,82                            
43261                                           2.128,81                            
43262                                           2.193,95                            
43263                                           2.197,71                            
43264                                           2.228,80                            

       ... Despesas Dedutíveis - Instrução [R$ milhões]  \
43260  ...                                          NaN   
43261  ...                                          NaN   
43262  ...                                          NaN   
43263  ...                                          NaN   
43264  ...                                          NaN   

      Despesas Dedutíveis - Médicas [R$ milhões]  \
43260                                        NaN   
43261                                        NaN   
43262                                        NaN   
43263                                        NaN   
43264                        

In [4]:
df20br = df20[df20["Ente Federativo"] == "BRASIL"]
df20br = df20br.reset_index()
df20br = df20br.drop(["Ente Federativo" , "index"], axis=1)
df20br.index = df20br.index + 1
df20br.head()

Centil  Quantidade de Contribuintes  \
1      1                      316.349   
2      2                      316.348   
3      3                      316.349   
4      4                      316.348   
5      5                      316.349   

  Rendimentos Tributaveis - Limite Superior da RTB do Centil [R$ milhões]  \
1                                                NaN                        
2                                                NaN                        
3                                                NaN                        
4                                                NaN                        
5                                                NaN                        

  Rendimentos Tributaveis - Soma da RTB do Centil [R$ milhões]  \
1                                                NaN             
2                                                NaN             
3                                                NaN             
4                                                NaN             
5                                                NaN             

  Rendimentos Tributaveis - RTB Acumulada do Centil [R$ milhões]  \
1                                                NaN               
2                                                NaN               
3                                                NaN               
4                                                NaN               
5                                                NaN               

  Rendimentos Tributaveis - Média da RTB do Centil [R$]  \
1                                                NaN      
2                                                NaN      
3                                                NaN      
4                                                NaN      
5                                                NaN      

  Rendimentos Sujeitos à Tribut. Exclusiva [R$ milhões]  \
1                                           2.292,22      
2                                           2.774,79      
3                                           2.735,77      
4                                           2.314,48      
5                                           2.378,27      

  Rendimentos Isentos - Lucros e dividendos [R$ milhões]  \
1                                           4.971,46       
2                                           5.276,95       
3                                           4.733,55       
4                                           4.838,44       
5                                           5.007,19       

  Rendimentos Isentos - Rendim. Sócio/Titular ME/EPP Opt SIMPLES [R$ milhões]  \
1                                           2.199,82                            
2                                           2.128,81                            
3                                           2.193,95                            
4                                           2.197,71                            
5                                           2.228,80                            

  Rendimentos Isentos - Outros Rendimentos Isentos [R$ milhões]  ...  \
1                                          10.656,53             ...   
2                                          10.672,99             ...   
3                                          10.207,91             ...   
4                                          10.789,54             ...   
5                                          10.527,97             ...   

  Despesas Dedutíveis - Instrução [R$ milhões]  \
1                                          NaN   
2                                          NaN   
3                                          NaN   
4                                          NaN   
5                                          NaN   

  Despesas Dedutíveis - Médicas [R$ milhões]  \
1                                        NaN   
2                                        NaN   
3                                        NaN   
4 

In [5]:
import numpy as np

def convert_brazilian_number(value):
    """
    Convert Brazilian-formatted number string to float
    - Removes dots used as thousand separators
    - Replaces comma with dot for decimal separation
    """
    if pd.isna(value):
        return np.nan
    
    value = value.replace('.', '')
    value = value.replace(',', '.')
    return float(value)

# Convert all columns in df20br
for column in df20br.columns:
    if df20br[column].dtype == "object":
        try:
            df20br[column] = df20br[column].apply(convert_brazilian_number)
        except Exception as e:
            print(f"Error converting column {column}: {e}")

# Verify the data types
df20br.dtypes

Centil                                                                         float64
Quantidade de Contribuintes                                                    float64
Rendimentos Tributaveis - Limite Superior da RTB do Centil [R$ milhões]        float64
Rendimentos Tributaveis - Soma da RTB do Centil [R$ milhões]                   float64
Rendimentos Tributaveis - RTB Acumulada do Centil [R$ milhões]                 float64
Rendimentos Tributaveis - Média da RTB do Centil [R$]                          float64
Rendimentos Sujeitos à Tribut. Exclusiva [R$ milhões]                          float64
Rendimentos Isentos - Lucros e dividendos [R$ milhões]                         float64
Rendimentos Isentos - Rendim. Sócio/Titular ME/EPP Opt SIMPLES [R$ milhões]    float64
Rendimentos Isentos - Outros Rendimentos Isentos [R$ milhões]                  float64
Despesas Dedutíveis - Previdência [R$ milhões]                                 float64
Despesas Dedutíveis - Dependentes [R$ milhõ

In [30]:
df_99centil = df20br[df20br['Centil'] < 100]

import matplotlib.pyplot as plt
import seaborn as sns

def plot_limite_superior_rendimentos_matplotlib(df):
    plt.figure(figsize=(12, 6))
    plt.bar(df.index, df['Rendimentos Tributaveis - Soma da RTB do Centil [R$ milhões]'])
    plt.title('Rendimentos Tributaveis por Centil', fontsize=15)
    plt.xlabel('Centil', fontsize=12)
    plt.ylabel('Rendimentos Tributaveis [R$ milhões]', fontsize=12)
    plt.tight_layout()
    plt.show()

def plot_limite_superior_rendimentos_seaborn(df):
    plt.figure(figsize=(12, 6))
    sns.barplot(x=df.index, y=df['Rendimentos Tributaveis - Soma da RTB do Centil [R$ milhões]'])
    plt.title('Rendimentos Tributaveis por Centil', fontsize=15)
    plt.xlabel('Centil', fontsize=12)
    plt.ylabel('Rendimentos Tributaveis [R$ milhões]', fontsize=12)
    plt.tight_layout()
    plt.show()

def plot_limite_superior_rendimentos_line_matplotlib(df):
    plt.figure(figsize=(12, 6))
    plt.plot(df.index, df['Rendimentos Tributaveis - Soma da RTB do Centil [R$ milhões]'], 
             marker='o', linestyle='-', linewidth=2, markersize=8)
    plt.title('Rendimentos Tributaveis por Centil', fontsize=15)
    plt.xlabel('Centil', fontsize=12)
    plt.ylabel('Rendimentos Tributaveis [R$ milhões]', fontsize=12)
    plt.grid(True, linestyle='--', alpha=0.7)
    plt.tight_layout()
    plt.show()

def plot_limite_superior_rendimentos(df):
    fig = px.bar(df, 
        x=df.index, 
        y='Rendimentos Tributaveis - Soma da RTB do Centil [R$ milhões]',
        title='Rendimentos Tributaveis por Centil',
        labels={
            'index': 'Centil',
            'Rendimentos Tributaveis - Soma da RTB do Centil': 'Rendimentos Tributaveis'
    })

    fig.update_layout(
        xaxis_title='Centil',
        yaxis_title='Rendimentos Tributaveis',
        template='plotly_white'
    )

    fig.show()

def plot_bar_limite_superior_rendimentos(_df):
    # Create a figure with bar and line traces
    fig = go.Figure()

    # Add bar graph for rendimentos tributaveis
    fig.add_trace(go.Bar(
        x=_df.index, 
        y=_df['Rendimentos Tributaveis - Soma da RTB do Centil [R$ milhões]'],
        name='Rendimentos Tributaveis',
        opacity=1
    ))

    fig.update_layout(
        title='Rendimentos Tributaveis por Centil',
        xaxis_title='Centil',
        yaxis_title='Rendimentos Tributaveis',
        template='plotly_white',
        barmode='overlay',
        showlegend=False
    )

    fig.show()

In [9]:
def create_one_indexed_df(_df):
    _df = _df.reset_index()
    _df = _df.drop(["index"], axis=1)
    _df.index = _df.index + 1
    return _df

In [35]:
# Filter the DataFrame to remove rows with centil values 100 and 1001 to 10010
df_all_info = df20br[(df20br['Centil'] != 100) & (df20br['Centil'] != 10010)]
df_all_info = create_one_indexed_df(df_all_info)
df_all_info

Centil  Quantidade de Contribuintes  \
1          1.0                      316.349   
2          2.0                      316.348   
3          3.0                      316.349   
4          4.0                      316.348   
5          5.0                      316.349   
..         ...                          ...   
114   100106.0                        3.164   
115   100107.0                        3.163   
116   100108.0                        3.164   
117   100109.0                        3.163   
118  1001010.0                        3.163   

     Rendimentos Tributaveis - Limite Superior da RTB do Centil [R$ milhões]  \
1                                                  NaN                         
2                                                  NaN                         
3                                                  NaN                         
4                                                  NaN                         
5                                                  NaN                         
..                                                 ...                         
114                                       1.475119e+06                         
115                                       1.754642e+06                         
116                                       2.264549e+06                         
117                                       3.520592e+06                         
118                                       3.307760e+08                         

     Rendimentos Tributaveis - Soma da RTB do Centil [R$ milhões]  \
1                                                  NaN              
2                                                  NaN              
3                                                  NaN              
4                                                  NaN              
5                                                  NaN              
..                                                 ...              
114                                            4371.27              
115                                            5075.86              
116                                            6267.54              
117                                            8755.18              
118                                           26439.14              

     Rendimentos Tributaveis - RTB Acumulada do Centil [R$ milhões]  \
1                                                  NaN                
2                                                  NaN                
3                                                  NaN                
4                                                  NaN                
5                                                  NaN                
..                                                 ...                
114                                           21172.52                
115                                           26248.38                
116                                           32515.92                
117                                           41271.10                
118                                           67710.24                

     Rendimentos Tributaveis - Média da RTB do Centil [R$]  \
1                                                  NaN       
2                                                  NaN       
3                                                  NaN       
4                                                  NaN       
5                                                  NaN       
..                                                 ...       
114                                         1381564.53       
115                                         1604759.97       
116                                         1980892.39       
117                                         2767997.25       
118                                         8358880.64       

     Rendimentos Sujeitos à Tribut. Exclusiva [R$ milhõ

In [71]:
import plotly.graph_objects as go
import numpy as np

def plot_limite_superior_rendimentos_custom(df):
    # Create a custom x-axis mapping
    def map_centil_x(centil):
        if centil <= 99:
            return centil
        elif 1001 <= centil <= 1009:
            return 100 + (centil - 1000) / 10
        elif 100101 <= centil <= 100110:
            return 100 + 1 + (centil - 100100) / 100
    
    # Create a custom width mapping
    def map_bar_width(centil):
        if centil <= 99:
            return 0.8  # Full width for main centils
        elif 1001 <= centil <= 1009:
            return 0.3  # Narrower for first decil of top centil
        elif 100101 <= centil <= 100110:
            return 0.1  # Very narrow for second decil of top centil
        else:
            return 0.8  # Default width
    
    # Create a copy of the dataframe with mapped x-axis and width
    df_mapped = df.copy()
    df_mapped['x_axis'] = df_mapped['Centil'].apply(map_centil_x)
    df_mapped['bar_width'] = df_mapped['Centil'].apply(map_bar_width)
    
    # Sort by the new x-axis to ensure correct order
    df_mapped = df_mapped.sort_values('x_axis')
    
    # Get the y-column
    y_column = 'Rendimentos Tributaveis - Limite Superior da RTB do Centil [R$ milhões]'
    
    # Print min/max values for debugging
    print(f"Min value: {df_mapped[y_column].min()}")
    print(f"Max value: {df_mapped[y_column].max()}")
    
    # Create two separate dataframes - one for regular values and one for the extreme value
    # Find the highest value (likely the top 0.01%)
    max_value = df_mapped[y_column].max()
    threshold = max_value * 0.5  # Adjust this threshold as needed
    
    df_regular = df_mapped[df_mapped[y_column] < threshold]
    df_extreme = df_mapped[df_mapped[y_column] >= threshold]
    
    # Create the Plotly figure
    fig = go.Figure()
    
    # Add bars for regular values
    fig.add_trace(go.Bar(
        x=df_regular['x_axis'], 
        y=df_regular[y_column],
        width=[width if not np.isnan(width) else 0.8 for width in df_regular['bar_width']],
        marker_color='rgba(33, 102, 172, 0.8)',  # Blue color
        marker_line_color='rgb(33, 102, 172)',
        marker_line_width=1.5,
        name='Regular Values',
        hovertemplate='Centil: %{x}<br>Rendimentos: %{y:.2f} R$ milhões<extra></extra>'
    ))
    
    # Add bars for extreme values with a different color
    if not df_extreme.empty:
        fig.add_trace(go.Bar(
            x=df_extreme['x_axis'], 
            y=df_extreme[y_column],
            width=[width if not np.isnan(width) else 0.8 for width in df_extreme['bar_width']],
            marker_color='rgba(255, 0, 0, 0.8)',  # Red color for emphasis
            marker_line_color='rgb(200, 0, 0)',
            marker_line_width=1.5,
            name='Top 0.01%',
            hovertemplate='Centil: %{x}<br>Rendimentos: %{y:.2f} R$ milhões<extra></extra>'
        ))
    
    # Create a broken axis effect with two subplots
    fig = go.Figure()
    
    # First subplot for regular values
    fig.add_trace(go.Bar(
        x=df_mapped['x_axis'], 
        y=df_mapped[y_column],
        width=[width if not np.isnan(width) else 0.8 for width in df_mapped['bar_width']],
        marker_color=['rgba(255, 0, 0, 0.8)' if y >= threshold else 'rgba(33, 102, 172, 0.8)' for y in df_mapped[y_column]],
        marker_line_color=['rgb(200, 0, 0)' if y >= threshold else 'rgb(33, 102, 172)' for y in df_mapped[y_column]],
        marker_line_width=1.5,
        hovertemplate='Centil: %{x}<br>Rendimentos: %{y:.2f} R$ milhões<extra></extra>'
    ))

    # Customize layout with explicit y-axis range
    fig.update_layout(
        title={
            'text': 'Rendimentos Tributaveis por Centil (Expanded)',
            'y':0.95,
            'x':0.5,
            'xanchor': 'center', 
            'yanchor': 'top',
            'font': dict(size=20)
        },
        xaxis_title='Centil',
        yaxis_title='Rendimentos Tributaveis [R$ milhões]',
        template='plotly_white',
        width=1200,  # Wider figure
        height=600,
        xaxis=dict(
            tickmode='array',
            tickvals=list(range(0, 101, 10)) + [100.1, 100.2, 100.3, 100.4, 100.5, 100.6, 100.7, 100.8, 100.9, 101],
            ticktext=[str(x) if x <= 100 else f'100.{int((x-100)*10)}' for x in list(range(0, 101, 10)) + [100.1, 100.2, 100.3, 100.4, 100.5, 100.6, 100.7, 100.8, 100.9, 101]],
            tickangle=-45
        ),
        # Set y-axis to auto-range by default
        yaxis=dict(
            tickformat='.2f',  # Two decimal places
            autorange=True
        )
    )

    # Add hover effects
    fig.update_traces(
        marker_line_width=1.5
    )

    # Add buttons for different scale options
    fig.update_layout(
        updatemenus=[
            dict(
                type="buttons",
                direction="right",
                buttons=list([
                    dict(
                        args=[{"yaxis.type": "linear", "yaxis.autorange": True}],
                        label="Linear Scale (Auto)",
                        method="relayout"
                    ),
                    dict(
                        args=[{"yaxis.type": "log", "yaxis.autorange": True}],
                        label="Log Scale",
                        method="relayout"
                    ),
                    dict(
                        args=[{"yaxis.type": "linear", "yaxis.range": [0, threshold * 1.1]}],
                        label="Focus on Lower 99.99%",
                        method="relayout"
                    ),
                    dict(
                        args=[{"yaxis.type": "linear", "yaxis.range": [threshold * 0.9, max_value * 1.1]}],
                        label="Focus on Top 0.01%",
                        method="relayout"
                    )
                ]),
                pad={"r": 10, "t": 10},
                showactive=True,
                x=0.1,
                xanchor="left",
                y=1.1,
                yanchor="top"
            ),
        ]
    )

    # Add annotations to highlight the extreme value
    if not df_extreme.empty:
        for idx, row in df_extreme.iterrows():
            fig.add_annotation(
                x=row['x_axis'],
                y=row[y_column],
                text=f"Top 0.01%: {row[y_column]:.2f}",
                showarrow=True,
                arrowhead=1,
                ax=0,
                ay=-40
            )

    fig.show()

# Usage
plot_limite_superior_rendimentos_custom(df_all_info)

Min value: 100.0
Max value: 330776031.92


In [67]:
plot_limite_superior_rendimentos_custom(df_filtered)

Min value: 100.0
Max value: 3520591.77


In [43]:
df_last_centil = df20br[(df20br['Centil'] > 100) & (df20br['Centil'] < 10010)]
df_last_centil = create_one_indexed_df(df_last_centil)

In [12]:
df_last_milezil_without_last = df20br[(df20br['Centil'] > 100000) & (df20br['Centil'] < 100110)]
df_last_milezil_without_last = create_one_indexed_df(df_last_milezil_without_last)

In [13]:
df_last_milezil = df20br[(df20br['Centil'] > 100000)]
df_last_milezil = create_one_indexed_df(df_last_milezil)

In [14]:
df_100centil = df20br[df20br['Centil'] < 101]
df_100centil = create_one_indexed_df(df_100centil)

In [15]:
indexes = np.arange(1, 101)

# Create an array of Y values with the same value repeated
y_values = np.ones(100)*100

fig_renda_igualitaria = px.line(x=indexes, y=y_values,
              title='Rendimentos Tributaveis Igualitários',
              labels={
                  'x': 'Centil',
                  'y': 'Renda Igualitária'
              })

# Customize the layout
fig_renda_igualitaria.update_layout(
    xaxis_title='Index',
    yaxis_title='Renda Igualitária',
    template='plotly_white'
)

## Análises

### Qual a parcela da população presente nos dados da Receita?

Estes são os dados da distribuição de renda do Brasil em 2020, do 1º ao 99º centil.

Pegue a população brasileira que pagou imposto de renda e divida em 100 partes iguais.

Isto é um centil.

O primeiro centil são as 316.349 pessoas de menor renda.

O último centil são as 316.349 pessoas de maior renda.

A Renda Tributável Bruta começa a aparecer no 7º centil - abaixo disso as pessoas estão isentas da tributação.

A pessoa com maior renda do 7º centil possui uma renda tributável bruta de 100 reais - ou seja, pagam imposto em cima de 100 reais.

Uma distribuição totalmente igualitária de renda seria uma linha reta, algo como isso:

In [16]:
fig_renda_igualitaria.show()

Removi o último centil (100) para melhorar a visualização.

A distribuição de renda em 2020 no Brasil se parece com isso:

In [17]:
plot_limite_superior_rendimentos(df_99centil)

In [18]:
plot_limite_superior_rendimentos(df_99centil)

Até o 80º centil, o limite superior cresce de forma quase linear.

No 80º centil, a renda superior é de 82 mil.

Do 80 ao 90 esse crescimento se acelera, ou seja, mais concentração de renda.

O limite superior do 90º centil é de 129 mil.

Mas do 90 em diante, ou seja, os 10% contribuintes mais ricos, o crescimento vira exponencial.

A concentração de renda se acentua.

O limite superior do 99º centil é de 392 mil de renda tributável bruta por ano.

Mas a Receita divide o último centil, os 1% mais ricos, em 10 parcelas - 10 decis.

Ou seja o 1% mais rico é dividido, do 0,99% mais rico ao 0,1% mais rico.

A distribuição é a seguinte:

In [19]:
plot_limite_superior_rendimentos(df_last_centil)

Novamente, excluí o 1% mais rico pra facilitar a visualização.

O 0,1% mais pobre dos 1% mais ricos possui o limite superior de renda de 405 mil anuais.

O último decil analisado (do 0,2% mais ricos até os 0,1%) possui renda de 904 mil anuais.

A Receita ainda divide os 0,1% mais ricos em novas 10 partes. Vejamos:

In [20]:
plot_limite_superior_rendimentos(df_last_milezil_without_last)

Novamente, excluí os 0,01% mais ricos.

A primeira parte dos 0,1% mais ricos ganha até 953 mil por ano - nada mal.

A penúltima parte, os 0,02% mais ricos, ganha até 3 milhões e 521 mil por ano.

E os 0,01%? Vamo lá:

In [21]:
plot_limite_superior_rendimentos(df_last_milezil)

Essa é a distribuição dos 0,1% mais ricos, incluindo os 0,01% mais ricos.

Os últimos nessa faixa não são pobres - ganham até 953 mil por ano.

O topo da pirâmide ganha 331 milhões por ano.

Quase uma ordem de magnitude maior.

A diferença é tão brutal, mesmo em relação aos outros no 0,1% mais rico, que o gráfico parece uma linha no 0.

Quase toda a renda está nos 0,01% mais ricos.

E como se parece o gráfico de todos os contribuintes, incluindo os 1% restantes?

In [22]:
plot_limite_superior_rendimentos(df_100centil)

Como se todo o resto não tivesse nenhuma renda tributável.

Toda a renda do país de 212 milhões de pessoas está concentrada em 3.163 pessoas.

Agora, me diga, a tributação dos 0,01% também se parece com isso?

## Como se comparam as rendas com tributos e dívidas?

## Como é a evolução da renda ao longo do tempo?